src/core/connection.c

```diff
@@ -4411,6 +4411,18 @@ QuicConnRecvFrames(
     uint16_t PayloadLength = Packet->PayloadLength;
     uint64_t RecvTime = CxPlatTimeUs64();
 
+    //
+    // RFC 9000 Section 10.2.1: 
+    // In closing state, respond to any packet with CONNECTION_CLOSE, rate-limited
+    // to once every 5ms.
+    //
+    if (Closed && !Connection->State.ShutdownComplete) {
+        if (RecvTime - Connection->LastCloseResponseTimeUs >= QUIC_CLOSING_RESPONSE_MIN_INTERVAL_US) {
+            QuicSendSetSendFlag(&Connection->Send, QUIC_CONN_SEND_FLAG_CONNECTION_CLOSE);
+            Connection->LastCloseResponseTimeUs = RecvTime;
```

Comment:
Shouldn't `LastCloseResponseTimeUs` when closing the connection?
This probably works as long as it has been zeroed out, but we might send a second CONNECTION_CLOSE right after the first instead of 5us later.